### PR TITLE
add helm chart for tiflow-operator

### PR DESCRIPTION
**Pull request by Ehco1996**
### What problem does this PR solve?
as title

Issue link:

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Check List  

Tests 

- [ ] Unit test
- [ ] E2E test
- [ ] Stability test
- [ ] Manual test (add detailed scripts or steps below)
deployed in eks provider
* `helm install`

```
❯ helm install ehco ./tiflow-operator --debug
install.go:178: [debug] Original chart version: ""
install.go:195: [debug] CHART PATH: /Users/ehco/Documents/pingcap/cloud/tiflow-operator/deploy/charts/tiflow-operator

client.go:128: [debug] creating 1 resource(s)
install.go:151: [debug] CRD tiflowclusters.pingcap.com is already present. Skipping.
client.go:128: [debug] creating 6 resource(s)
NAME: ehco
LAST DEPLOYED: Thu Sep  1 10:59:11 2022
NAMESPACE: tiflow-operator-system
STATUS: deployed
REVISION: 1
TEST SUITE: None
USER-SUPPLIED VALUES:
{}

COMPUTED VALUES:
image:
  pullPolicy: IfNotPresent
  repository: steponce7/controller
  tag: latest
replicaCount: 1
resources:
  limits:
    cpu: 500m
    memory: 128Mi
  requests:
    cpu: 10m
    memory: 64Mi
service:
  port: 80
  type: ClusterIP
serviceAccount:
  annotations: {}
  create: true
  name: ""

HOOKS:
MANIFEST:
---
# Source: tiflow-operator/templates/manager-rbac.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: ehco-tiflow-operator-manager
  namespace: tiflow-operator-system
  labels:
    helm.sh/chart: tiflow-operator-0.1.0
    app.kubernetes.io/name: tiflow-operator
    app.kubernetes.io/instance: ehco
    app.kubernetes.io/version: "1.16.0"
    app.kubernetes.io/managed-by: Helm
---
# Source: tiflow-operator/templates/manager-rbac.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  labels:
    app: ehco-tiflow-operator-manager-role
  name: ehco-tiflow-operator-manager-role
rules:
- apiGroups:
  - ""
  resources:
  - configmaps
  - persistentvolumeclaims
  - persistentvolumes
  - secrets
  - services
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - ""
  resources:
  - pods
  verbs:
  - delete
  - get
  - list
  - update
  - watch
- apiGroups:
  - apps
  resources:
  - deployments
  - statefulsets
  verbs:
  - '*'
- apiGroups:
  - networking.k8s.io
  resources:
  - ingresses
  verbs:
  - '*'
- apiGroups:
  - pingcap.com
  resources:
  - standalones
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - pingcap.com
  resources:
  - standalones/finalizers
  verbs:
  - update
- apiGroups:
  - pingcap.com
  resources:
  - standalones/status
  verbs:
  - get
  - patch
  - update
- apiGroups:
  - pingcap.com
  resources:
  - tiflowclusters
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - pingcap.com
  resources:
  - tiflowclusters/finalizers
  verbs:
  - update
- apiGroups:
  - pingcap.com
  resources:
  - tiflowclusters/status
  verbs:
  - get
  - patch
  - update
- apiGroups:
  - storage.k8s.io
  resources:
  - storageclasses
  verbs:
  - get
  - list
  - watch
---
# Source: tiflow-operator/templates/manager-rbac.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  labels:
    app: ehco-tiflow-operator
  name: ehco-tiflow-operator-manager-role-binding
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: ehco-tiflow-operator-manager-role
subjects:
  - kind: ServiceAccount
    name: ehco-tiflow-operator-manager
    namespace: tiflow-operator-system
---
# Source: tiflow-operator/templates/manager-rbac.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  name: ehco-tiflow-operator-leader-election-role
rules:
- apiGroups:
  - ""
  resources:
  - configmaps
  verbs:
  - get
  - list
  - watch
  - create
  - update
  - patch
  - delete
- apiGroups:
  - coordination.k8s.io
  resources:
  - leases
  verbs:
  - get
  - list
  - watch
  - create
  - update
  - patch
  - delete
- apiGroups:
  - ""
  resources:
  - events
  verbs:
  - create
  - patch
---
# Source: tiflow-operator/templates/manager-rbac.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  name: ehco-tiflow-operator-leader-election-rolebinding
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: Role
  name: ehco-tiflow-operator-leader-election-role
subjects:
- kind: ServiceAccount
  name: ehco-tiflow-operator-manager
  namespace: tiflow-operator-system
---
# Source: tiflow-operator/templates/manager-deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: ehco-tiflow-operator
  namespace: tiflow-operator-system
  labels:
    helm.sh/chart: tiflow-operator-0.1.0
    app.kubernetes.io/name: tiflow-operator
    app.kubernetes.io/instance: ehco
    app.kubernetes.io/version: "1.16.0"
    app.kubernetes.io/managed-by: Helm
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/name: tiflow-operator
      app.kubernetes.io/instance: ehco
  template:
    metadata:
      labels:
        app.kubernetes.io/name: tiflow-operator
        app.kubernetes.io/instance: ehco
    spec:
      securityContext:
        runAsNonRoot: true
      serviceAccountName: ehco-tiflow-operator-manager
      containers:
        - name: manager
          securityContext:
            runAsNonRoot: true
          image: "steponce7/controller:latest"
          imagePullPolicy: IfNotPresent
          securityContext:
            allowPrivilegeEscalation: false
            capabilities:
              drop:
                - "ALL"
          livenessProbe:
            httpGet:
              path: /healthz
              port: 8081
            initialDelaySeconds: 15
            periodSeconds: 20
          readinessProbe:
            httpGet:
              path: /readyz
              port: 8081
            initialDelaySeconds: 5
            periodSeconds: 10
          resources:
            limits:
              cpu: 500m
              memory: 128Mi
            requests:
              cpu: 10m
              memory: 64Mi
          command:
            - /manager
          args:
            - --leader-elect

```

```
❯ k logs ehco-tiflow-operator-5cdc459b4-qsv64 -f
I0901 02:59:25.569990       1 request.go:601] Waited for 1.032920323s due to client-side throttling, not priority and fairness, request: GET:https://172.20.0.1:443/apis/storage.k8s.io/v1beta1?timeout=32s
1.6620011655735004e+09	INFO	controller-runtime.metrics	Metrics server is starting to listen	{"addr": ":8080"}
1.6620011655739653e+09	INFO	setup	starting manager
1.662001165574274e+09	INFO	Starting server	{"path": "/metrics", "kind": "metrics", "addr": "[::]:8080"}
1.6620011655743363e+09	INFO	Starting server	{"kind": "health probe", "addr": "[::]:8081"}
I0901 02:59:25.574474       1 leaderelection.go:248] attempting to acquire leader lease tiflow-operator-system/56a1d92f.pingcap.com...
I0901 02:59:40.744165       1 leaderelection.go:258] successfully acquired lease tiflow-operator-system/56a1d92f.pingcap.com
1.6620011807443445e+09	DEBUG	events	Normal	{"object": {"kind":"Lease","namespace":"tiflow-operator-system","name":"56a1d92f.pingcap.com","uid":"fdea6a01-79b3-4da6-980f-80e799ff8da2","apiVersion":"coordination.k8s.io/v1","resourceVersion":"10287462"}, "reason": "LeaderElection", "message": "ehco-tiflow-operator-5cdc459b4-qsv64_a03e2834-9fb0-42ab-8144-a72fc085a2a0 became leader"}
1.6620011807448473e+09	INFO	Starting EventSource	{"controller": "tiflowcluster", "controllerGroup": "pingcap.com", "controllerKind": "TiflowCluster", "source": "kind source: *v1alpha1.TiflowCluster"}
1.6620011807448819e+09	INFO	Starting EventSource	{"controller": "tiflowcluster", "controllerGroup": "pingcap.com", "controllerKind": "TiflowCluster", "source": "kind source: *v1.StatefulSet"}
1.6620011807448878e+09	INFO	Starting EventSource	{"controller": "tiflowcluster", "controllerGroup": "pingcap.com", "controllerKind": "TiflowCluster", "source": "kind source: *v1.Deployment"}
1.6620011807448988e+09	INFO	Starting EventSource	{"controller": "tiflowcluster", "controllerGroup": "pingcap.com", "controllerKind": "TiflowCluster", "source": "kind source: *v1.Service"}
1.662001180744903e+09	INFO	Starting EventSource	{"controller": "tiflowcluster", "controllerGroup": "pingcap.com", "controllerKind": "TiflowCluster", "source": "kind source: *v1.ConfigMap"}
1.6620011807449114e+09	INFO	Starting Controller	{"controller": "tiflowcluster", "controllerGroup": "pingcap.com", "controllerKind": "TiflowCluster"}
1.6620011810490572e+09	INFO	Starting workers	{"controller": "tiflowcluster", "controllerGroup": "pingcap.com", "controllerKind": "TiflowCluster", "worker count": 1}
```

- [x] No code

Side effects

- [ ] Breaking backward compatibility
- [ ] Other side effects:

### CODE REVIEW

File: deploy/charts/tiflow-operator/templates/manager-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  template:
+    metadata:
+      labels:
+        {{- include "tiflow-operator.selectorLabels" . | nindent 8 }}
+    spec:
+      securityContext:
+        runAsNonRoot: true
+      serviceAccountName: {{ include "tiflow-operator.serviceAccountName" . }}-manager
+      containers:
+        - name: manager
```
Comment (Ehco1996) [inline deploy/charts/tiflow-operator/templates/manager-deployment.yaml:22]: i removed the  `kube-rbac-proxy` contiainer  releated to https://github.com/pingcap/tiflow-operator/issues/26

**Comment by Ehco1996** [comment]
waiting https://github.com/pingcap/tiflow-operator/pull/25 merged first, because role/rolebingd has changed

File: deploy/charts/tiflow-operator/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.1.0
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+appVersion: "1.16.0"
```
Comment (lichunzhu) [inline deploy/charts/tiflow-operator/Chart.yaml]: why `1.16.0`?

File: deploy/charts/tiflow-operator/templates/manager-rbac.yaml [hunk 3]
```diff
@@ -0,0 +1,187 @@
+{{- if .Values.serviceAccount.create -}}
```
Comment (lichunzhu) [inline deploy/charts/tiflow-operator/templates/manager-rbac.yaml:1]: if code is updated, do we need to rewrite this rbac file manually?

File: deploy/charts/tiflow-operator/Chart.yaml (hunk 2 above)
Comment (Ehco1996) [reply deploy/charts/tiflow-operator/Chart.yaml]: this is generated by `helm` ,maybe i should  rename to `v1-canary` like [tidb operator](https://github.com/pingcap/tidb-operator/blob/b84b18b838df29f17f0020b42fa72c6bda3e8b77/charts/tidb-operator/Chart.yaml#L4)?

File: deploy/charts/tiflow-operator/templates/manager-rbac.yaml (hunk 3 above)
Comment (Ehco1996) [reply deploy/charts/tiflow-operator/templates/manager-rbac.yaml:1]: i am sorry but yes, unless `kustomize` support generate charts as well

File: deploy/charts/tiflow-operator/Chart.yaml (hunk 2 above)
Comment (lichunzhu) [reply deploy/charts/tiflow-operator/Chart.yaml]: I prefer `v1-canary`.

Comment (Ehco1996) [reply deploy/charts/tiflow-operator/Chart.yaml]: fixed in https://github.com/pingcap/tiflow-operator/pull/28/commits/9d8b753984e5d96d051275d5b7732db1d0ef731c

Comment (Ehco1996) [reply deploy/charts/tiflow-operator/Chart.yaml]: ping @lichunzhu

**Review by lichunzhu — APPROVED** [review]
LGTM

**Pull request merged**